api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -465,7 +466,54 @@ message HttpConnectionManager {
+  //    <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.original_ip_detection_extensions>`
+  //    should be used to configure the :ref:`xff extension <envoy_api_msg_extensions.http.original_ip_detection.xff.v3.XffConfig>`
+  //    to configure IP detection using the :ref:`config_http_conn_man_headers_x-forwarded-for` header. To replace
+  //    this field use a config like the following:
+  //
+  // .. code-block:: yaml
+  //
+  //    original_ip_detection_extensions:
+  //      typed_config:
+  //        "@type": type.googleapis.com/envoy.extensions.http.original_ip_detection.xff.v3.XffConfig
+  //        xff_num_trusted_hops: 1
+  //
+  uint32 xff_num_trusted_hops = 19
+      [deprecated = true, (envoy.annotations.deprecated_at_minor_version) = "3.0"];
+
+  // The configuration for the original-IP detection extensions.
+  //
+  // By default, when the :ref:`use_remote_address
+  // <envoy_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.use_remote_address>`
+  // option is not used, Envoy will rely on the :ref:`config_http_conn_man_headers_x-forwarded-for` header
+  // and the deprecated :ref:`xff_num_trusted_hops
+  // <envoy_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.xff_num_trusted_hops>`
+  // configuration option to detect the remote address. Once :ref:`xff_num_trusted_hops
+  // <envoy_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.xff_num_trusted_hops>`
+  // is removed, leaving this field empty will retain the default behavior when that field is unset. See
+  // the :ref:`xff extension <envoy_api_msg_extensions.http.original_ip_detection.xff.v3.XffConfig>` for how to
+  // configure xff_num_trusted_hops when using extensions.
+  //
+  // Furthermore, relying on the :ref:`config_http_conn_man_headers_x-forwarded-for`
+  // header doesn't necessarily work for all network configurations. For instance,
+  // if the number of trusted hops isn't consistent for requests flowing into
+  // Envoy this mechanism cannot be used reliably. For such cases, the following
+  // option allows for a list of custom extensions to be provided.
+  //
+  // When configured the extensions will be called along with the request headers
+  // and information about the downstream connection, such as direct connected address,
+  // when the HCM needs to populate the final remote address. Each extension will then rely
+  // on these parameters to decide the value of the request's final remote address.
+  // If an extension fails to detect the original IP address and isn't configured to reject
+  // the request when this happens, the HCM will then try the remaining configured extensions
+  // until one succeeds or the request is rejected. If the request isn't rejected nor any extension
+  // succeeds, the HCM will fallback to using the remote address.
```

Comment:
Which remote address? :) I'm guessing the socket level source IP?

Reply:
Yeah -- the HCM's. I'll make it explicit.

Reply:
Ok, after the trimming and rewording, nothing else might be needed here since at the start of the (new paragraph) I refer to the directly connected address.